osmoutils/cosmwasm/helpers.go

```diff
@@ -118,11 +120,16 @@ func Sudo[T any, K any](ctx sdk.Context, contractKeeper ContractKeeper, contract
 		return response, err
 	}
 
-	responseBz, err := contractKeeper.Sudo(ctx, sdk.MustAccAddressFromBech32(contractAddress), bz)
+	// Make contract call with a gas limit of 30M to ensure contracts cannot run unboundedly
+	childCtx := ctx.WithGasMeter(sdk.NewGasMeter(DefaultContractCallGasLimit))
```

Comment:
```suggestion
	childCtx := ctx.WithGasMeter(sdk.NewGasMeter(min(ctx.GasMeter().Limit(), DefaultContractCallGasLimit)))
```

(min requires go 1.21 :( and we probably need to make the default have type Gas, but you get the idea)

Reply:
Good point, added! dae3b329e980f38f2dad3150c87589c5d755639f

I just manually did the min check and set the gas limit to be of type `sdk.Gas`. Lmk if you see a better/simpler way to do it.

Reply:
no, that's probably the right way. Though I'd consider raising the min go version to 1.21 just to have generic min/max

Reply:
modified this to use min() now that we're on 1.21